crates/pop-cli/src/commands/install/mod.rs

```diff
@@ -2,32 +2,234 @@
+	cmd("export", vec!["OPENSSL_INCLUDE_DIR='/usr/include/openssl-1.0'"]).run()?;
+
+	Ok(())
+}
+
+async fn install_ubuntu(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/linux/")?;
+	if !skip_confirm {
+		prompt_for_confirmation("git, clang, curl, libssl-dev, protobuf-compiler")?
+	}
+	cmd(
+		"apt",
+		vec!["install", "--assume-yes", "git", "clang", "curl", "libssl-dev", "protobuf-compiler"],
+	)
+	.run()?;
+
+	Ok(())
+}
+
+async fn install_debian(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/linux/")?;
+	if !skip_confirm {
+		prompt_for_confirmation("cmake, pkg-config, libssl-dev, git, gcc, build-essential, git, protobuf-compiler, clang, libclang-dev")?
+	}
+	cmd(
+		"apt",
+		vec![
+			"install",
+			"-y",
+			"cmake",
+			"pkg-config",
+			"libssl-dev",
+			"git",
+			"gcc",
+			"build-essential",
+			"git",
+			"protobuf-compiler",
+			"clang",
+			"libclang-dev",
+		],
+	)
+	.run()?;
+
+	Ok(())
+}
+
+async fn install_redhat(skip_confirm: bool) -> anyhow::Result<()> {
+	log::info("More information about the packages to be installed here: https://docs.substrate.io/install/linux/")?;
+	if !skip_confirm {
+		prompt_for_confirmation("cmake, openssl-devel, git, protobuf, protobuf-compiler, clang, clang-devel and srustup")?;
+	}
+	cmd("yum", vec!["update", "-y"]).run()?;
+	cmd("yum", vec!["groupinstall", "-y", "'Development Tool"]).run()?;
+	cmd(
+		"yum",
+		vec![
+			"install",
+			"-y",
+			"cmake",
+			"openssl-devel",
+			"git",
+			"protobuf",
+			"protobuf-compiler",
+			"clang",
+			"clang-devel",
+		],
+	)
+	.run()?;
+
+	Ok(())
+}
+
+fn prompt_for_confirmation(message: &str) -> anyhow::Result<()> {
+	if !confirm(format!(
+		"📦 Do you want to proceed with the installation of the following packages: {} ?",
+		message
+	))
+	.initial_value(true)
+	.interact()?
+	{
+		return Err(anyhow::anyhow!("🚫 You have cancelled the installation process."));
+	}
+	Ok(())
+}
+
+fn not_supported_message() -> anyhow::Result<()> {
+	log::error("This OS is not supported at present")?;
+	log::warning("⚠️ Please refer to https://docs.substrate.io/install/ for setup information.")?;
+	Ok(())
+}
+
+async fn install_rustup() -> anyhow::Result<()> {
+	match cmd("which", vec!["rustup"]).read() {
+		Ok(output) => {
+			log::info(format!("ℹ️ rustup installed already at {}.", output))?;
+			cmd("rustup", vec!["update"]).run()?;
+			cmd("rustup", vec!["default", "stable"]).run()?;
+		},
+		Err(_) => {
+			let mut spinner = cliclack::spinner();
+			spinner.start("Installing rustup ...");
+			run_external_script("https://sh.rustup.rs").await?;
+			outro("rustup installed!")?;
+			cmd(
+				"source",
+				vec![
+					"~/.cargo/env
+",
+				],
+			)
+			.run()?;
+			cmd("rustup", vec!["default", "stable"]).run()?;
+		},
+	}
+	cmd("rustup", vec!["update", "nightly"]).run()?;
```

Comment:
Note that the toolchain files should sort all this automatically on first build of a project, provided one exists. Fine to leave as is tho.